source/plugins/ruby/kubelet_utils.rb

```diff
@@ -17,11 +18,14 @@ def get_node_capacity
         cpu_capacity = 1.0
         memory_capacity = 1.0
 
-        response = CAdvisorMetricsAPIClient.getNodeCapacityFromCAdvisor(winNode: nil)
+        response = CAdvisorMetricsAPIClient.getAllMetricsCAdvisor(winNode: nil)
         if !response.nil? && !response.body.nil?
-          cpu_capacity = JSON.parse(response.body)["num_cores"].nil? ? 1.0 : (JSON.parse(response.body)["num_cores"] * 1000.0)
-          memory_capacity = JSON.parse(response.body)["memory_capacity"].nil? ? 1.0 : JSON.parse(response.body)["memory_capacity"].to_f
-          @log.info "CPU = #{cpu_capacity}mc Memory = #{memory_capacity / 1024 / 1024}MB"
+          all_metrics = response.body.split("\n")
+          cpu_capacity = all_metrics.select{|m| m.start_with?('machine_cpu_cores') }.first.split.last.to_f * 1000
```

Comment:
better to check the first element after the spilt is indeed 'machine_cpu_cores' for future proofing, so newer metrics with this prefix doesn't break us.

Reply:
Done